csrc/cpu/neighbor_sample_cpu.cpp

```diff
@@ -25,10 +34,15 @@ sample(const torch::Tensor &colptr, const torch::Tensor &row,
   auto *row_data = row.data_ptr<int64_t>();
   auto *input_node_data = input_node.data_ptr<int64_t>();
 
+  // Insert nodes to sample and populate local node map
+  auto input_i = 0;
   for (int64_t i = 0; i < input_node.numel(); i++) {
     const auto &v = input_node_data[i];
-    samples.push_back(v);
-    to_local_node.insert({v, i});
+    if (to_local_node.find(v) == to_local_node.end()){
```

Comment:
I personally think this is less efficient since we actually access and try to find elements in `to_local_node` twice. How about:
```c++
const auto res = to_local_node.insert({v, samples.size()});
if (res.second)
    samples.push_back(v);
```

Reply:
cool, much nicer, thanks!